tests/pipeline/test_blaze.py

```diff
@@ -271,32 +270,56 @@ def test_non_numpy_field(self):
         )
         with self.assertRaises(AttributeError):
             ds.a
-        self.assertIsInstance(object.__getattribute__(ds, 'a'), NonNumpyField)
+        self.assertIsInstance(
+            object.__getattribute__(ds, 'a'),
+            NonPipelineField,
+        )
 
-    def test_non_pipeline_field(self):
-        # NOTE: This test will fail if we ever allow string types in
-        # the Pipeline API. If this happens, change the dtype of the `a` field
-        # of expr to another type we don't allow.
+    def test_cols_with_missing_vals(self):
+        dates = (pd.Timestamp('2014-01-01'), pd.Timestamp('2014-01-03'))
+        df = pd.DataFrame({
+            'sid': self.sids[:-1],
+            'value': (0., 1.,),
+            'str_value': ("a", "b",),
+            'asof_date': dates,
+            'timestamp': dates,
+        })
         expr = bz.data(
-            [],
+            df,
             dshape="""
             var * {
-                 a: string,
+                 sid: int64,
+                 value: float64,
+                 str_value: string,
                  asof_date: datetime,
                  timestamp: datetime,
             }""",
         )
-        ds = from_blaze(
-            expr,
-            loader=self.garbage_loader,
-            no_deltas_rule=no_deltas_rules.ignore,
-        )
-        with self.assertRaises(AttributeError):
-            ds.a
-        self.assertIsInstance(
-            object.__getattribute__(ds, 'a'),
-            NonPipelineField,
-        )
+        fields = OrderedDict(expr.dshape.measure.fields)
+
+        with tmp_asset_finder() as finder:
+            expected = pd.DataFrame(
+                np.array([["a", 0],
+                          [None, np.NaN],
+                          [None, np.NaN],
+                          ["a", 0],
+                          [None, np.NaN],
+                          [None, np.NaN],
+                          ["a", 0],
+                          ["b", 1],
+                          [None, np.NaN]]),
+                columns=['str_value', 'value'],
+                index=pd.MultiIndex.from_product(
+                    (self.dates, finder.retrieve_all(self.sids))
```

Comment:
Nothing, really. All the tests in this suite assume the sids are A, B, and C, which are the sids used in `tmp_asset_finder`. We would probably want to update all of them, then.